doc/nrf/ug_bootloader.rst

```diff
@@ -7,212 +7,192 @@ Secure bootloader chain
+  This is useful when a bootloader performs additional complex operations, like using a software stack.
 
-   Memory layout
+  The following image shows an abstract representation of the memory layout for an application that uses both an immutable and an upgradable bootloader in its boot chain:
 
-For detailed information about the memory layout, see the partition configuration in the :file:`partitions.yml` file (located in the build folder directory) or run ``ninja partition_manager_report``.
+  .. figure:: images/bootloader_memory_layout.svg
+     :alt: Memory layout
+
+For detailed information about the memory layout, see the partition configuration in the :file:`partitions.yml` file, located in the build folder directory, or run ``ninja partition_manager_report``.
 
 .. note::
-   The Partition Manager must be enabled to make the :file:`parititions.yml` file and the partition_manager_report target available.
+   You must enable the Partition Manager to make the :file:`partitions.yml` file and the ``partition_manager_report`` target available.
    See :ref:`partition_manager` and :ref:`ug_multi_image` for details.
 
+By default, building an application with any bootloader configuration creates a :ref:`multi-image build <ug_multi_image>`, where the :ref:`partition_manager` manages its memory partitions.
+When building with a non-secure application, the :ref:`secure_partition_manager` is also built with the image automatically.
+
 .. _immutable_bootloader:
 
-Immutable bootloader
-====================
+First-stage Immutable bootloader
+================================
 
-The first step in the chain of trust is an immutable first stage bootloader, in the following referred to as immutable bootloader.
-The code abbreviates this bootloader as SB (Secure Boot, Secure Bootloader) or B0.
+The first step in the chain of trust is a secure, immutable, first-stage bootloader.
 
-The immutable bootloader runs after every reset and establishes the root of trust by verifying the signature and metadata of the next image in the boot sequence.
+This immutable bootloader runs after every reset and establishes the root of trust by verifying the signature and metadata of the next image in the boot sequence.
 If the verification fails, the boot process stops.
-
-This way, the immutable bootloader can guarantee that if the flash of the next image in the boot sequence (another bootloader or an application) has been tampered with in any way, the image will not start up.
-For example, if an attacker attempts to take over the device by altering the flash, the device will not boot, and thus not run the infected code.
+This way, the immutable bootloader can guarantee that the next image in the boot sequence will not start up if it has been tampered with in any way.
+For example, if an attacker attempts to take over the device by altering the firmware, the device will not boot, and thus not run the infected code.
```

Comment:
There is one important thing to note here. I guess it is obvious but for this to work bootloader must be truly immutable.
In the next paragraph it states

> immutable bootloader cannot be updated or deleted from the device unless you erase the device

The whole point however is that bootloader should not be modifiable in any way and that means that erase must be locked out (in HW if possible).

If one can do something to erase and change the bootloader (e.g. by connecting the debug pins), it means the whole chain of trust does not exist.

If process of locking out the flash is mentioned in some place then link to it should be provided. If it is not I believe it should be described.

Reply:
The locked flash is mentioned in the NSIB README, which I've now linked along with some wording changes.